src/broker/boot_config.c

```diff
@@ -235,6 +235,49 @@ int boot_config_parse (const flux_conf_t *cf,
     return 0;
 }
 
+int boot_config_attr (attr_t *attrs, json_t *hosts)
+{
+    struct hostlist *hl = NULL;
+    char *s = NULL;
+    size_t index;
+    json_t *value;
+    int rv = -1;
+
+    if (!hosts || json_array_size (hosts) == 0)
+        return 0;
+
+    if (!(hl = hostlist_create ()))
+        goto error;
+
+    json_array_foreach (hosts, index, value) {
+        const char *host;
+        if (json_unpack (value, "{s:s}", "host", &host) < 0) {
+            log_msg ("Internal error [bootstrap]: missing host field");
+            errno = EINVAL;
+            goto error;
+        }
+        if (hostlist_append (hl, host) < 0)
+            goto error;
+    }
+
+    if (!(s = hostlist_encode (hl)))
+        goto error;
+
+    if (attr_add (attrs,
+                  "config.hostlist",
+                  s,
+                  FLUX_ATTRFLAG_READONLY) < 0) {
```

Comment:
Should probably use FLUX_ATTRFLAG_IMMUTABLE here instead (see definitions in attr.h).  This would allow the value to be cached, since it can never change.

Reply:
i debated over this, but now that I think about it for another 10 seconds, IMMUTABLE sounds obvious.